### ISSUE TITLE
ui_test are failing

### ISSUE THREAD

**Issue by gsabran** [airbnb/BuckSample#18]
`make ui_test` fails for me. Should they be part of the ci ?

```
make ui_test
/usr/local/Cellar/buck/2018.10.29.01/bin/buck test //App/UITests:UITests
FAILURE BuckSampleUITests -[BuckSampleUITests testExample]: <unknown>:0: failed: caught "NSInternalInconsistencyException", "No target application path specified via test configuration: <XCTestConfiguration: 0x600003368270>
```

**Comment by dfed** [comment]
It [looks like] `xctool` isn't capable of running UI Tests. I'll take a task to investigate having `make ui_test` generate an Xcode project whose UI Test target is then run via `xcodebuild`

**Comment by mrtom** [comment]
There's another (possibly related?) issue with the UI Tests. Even tho `test_host_app` seems to be set correctly in the BUCK file for UITests, if you view the UITests target in Xcode* you'll see the host app is set to none and can't be changed - when it should be set to `ExampleAppBundle`.

This indicates to me that Buck isn't creating a valid Xcode project file for this target for some reason, although I've failed to figure out how/why.

![screenshot 2019-03-07 at 16 54 47](https://user-images.githubusercontent.com/174462/53974557-b8539d80-40fa-11e9-9202-0058c54f0c48.png)

*you'll have to re-enable the UITests by adding them to the `app_tests` array in the main BUCK file to get them to show up back in Xcode.

**Comment by dfed** [comment]
It kinda seems like neither `buck` nor `xctool` are set up to handle UI Tests 😬.

For what it's worth, internally we run screenshot tests using Buck's CLI using a host app, but we don't use Apple's UI testing framework to do so.

**Comment by mrtom** [comment]
Thanks that's useful to know!

I wonder if @milend will get a ping if I mention him and comment on 'buck not set up to handle UI tests'?

**Comment by qyang-nj** [comment]
The "Host Application is None" problem will be fixed by #62, it's because `.test` suffix breaks Buck hacks.

**Comment by dfed** [comment]
@qyang-nj indeed! One problem down.

That said, UI tests (i.e. tests that run `XCUIApplication().launch()`) will still crash per my local testing. But we should be able to run tests in a host app, and grab screenshots in that host app without running `XCUIApplication().launch()`.
